src/client.rs

```diff
@@ -827,7 +827,7 @@ where
 
                 // Safe to unwrap because we know this message has a certain length and has the code
                 // This reads the first byte without advancing the internal pointer and mutating the bytes
-                let code = *message.get(0).unwrap() as char;
+                let code = *message.get(0).unwrap_or(&('$' as u8)) as char;
```

Comment:
Doesn't look like this is safe to unwrap after all. Why `$`?

Reply:
I asked for this to be extra cautious, our old implementation would panic if for some reason we didn't have bytes in the message. The `$` is just to hit the last branch in the match statement and avoid the panic. 

The comment about this being safe is because `read_message` wouldn't return success unless the entire payload is loaded in the buffer
https://github.com/levkk/pgcat/blob/main/src/messages.rs#L469-L497

We could go without it if it looks confusing

Reply:
This feels like something we'd do in Ruby :) Let's be a bit more intentional in this mission-critical code.